packages/apps/src/app.events.ts

```diff
@@ -24,6 +24,30 @@ export function event<Name extends keyof IEvents>(
   return this;
 }
 
+type PluginWithEvents<TPlugin extends IPlugin> =
+  TPlugin extends IPluginWithEvents<infer TEvents> ? TEvents : never;
+
+type PluginConstructor = { new (...args: any[]): IPlugin & PluginWithEvents<any> };
+type PluginInstance = IPlugin & PluginWithEvents<any>;
+
+export function pluginEvent<
```

Comment:
the `app.event` that exists is where plugin events get emitted today, we should keep using that method to subscribe to plugin events. I think by implementing a generic for event types we can make this more implicit so that when plugins are imported the app automatically infers the events needed for the `app.event` method intellisense.

Reply:
Let me try once more to have these two types of events be merged :)